app/[eventSlug]/session-block.tsx

```diff
@@ -106,28 +107,45 @@ export function RealSessionCard(props: {
 }) {
   const { eventName, session, numHalfHours, location, guests, rsvpd } = props;
   const { user: currentUser } = useContext(UserContext);
-  const { updateRsvp, localSessions } = useContext(EventContext);
+  const { updateRsvp, localSessions, userBusySessions } =
+    useContext(EventContext);
   const router = useRouter();
   const [isRsvping, setIsRsvping] = useState(false);
 
   const hostStatus = currentUser && session.Hosts?.includes(currentUser);
   const lowerOpacity = !rsvpd && !hostStatus;
   const formattedHostNames = session["Host name"]?.join(", ") ?? "No hosts";
-  const [rsvpModalOpen, setRsvpModalOpen] = useState(false);
+  const [userModalOpen, setUserModalOpen] = useState(false);
+  const [clashingSession, setClashingSession] = useState(newEmptySession());
+  const [confirmRSVPModalOpen, setConfirmRSVPModalOpen] = useState(false);
   const screenWidth = useScreenWidth();
   const onMobile = screenWidth < 640;
 
   const handleClick = async () => {
     if (hostStatus) {
       return;
     }
-    if (currentUser && !onMobile) {
-      setIsRsvping(true);
-      await updateRsvp(currentUser, session.ID, rsvpd);
-      setIsRsvping(false);
+
+    const overlappingSession = userBusySessions.find((ses) =>
+      sessionsOverlap(session, ses)
+    );
+    if (!rsvpd && overlappingSession) {
+      setClashingSession(overlappingSession);
+      setConfirmRSVPModalOpen(true);
+    } else if (currentUser && !onMobile) {
+      await doRsvp();
     } else {
-      setRsvpModalOpen(true);
+      setUserModalOpen(true);
+    }
+  };
+
+  const doRsvp = async () => {
+    if (!currentUser) {
+      return;
     }
+    setIsRsvping(true);
+    await updateRsvp(currentUser, session.ID, rsvpd);
```

Comment:
We should avoid `await` in clientside code as much as possible. This should be refactored to 

`updateRsvp(currentUser, session.ID, rsvpd).then(setIsRsvping(false);`

and also remove the await in the onClick handler